Formula/hashlink.rb

```diff
@@ -30,10 +29,18 @@ class Hashlink < Formula
   end
 
   def install
-    inreplace "Makefile", /\$\{LFLAGS\}/, "${LFLAGS} ${EXTRA_LFLAGS}" unless build.head?
-    # On Linux, also pass EXTRA_FLAGS to LIBFLAGS, so that the linker will also add the RPATH to .hdll files.
-    inreplace "Makefile", "LIBFLAGS =", "LIBFLAGS = ${EXTRA_LFLAGS}"
-    system "make", "EXTRA_LFLAGS=-Wl,-rpath,#{libexec}/lib"
+    # remove with 1.13 release:
+    inreplace "Makefile", /PCRE_INCLUDE =/, "PCRE_FLAGS =" unless build.head?
+
+    if OS.mac?
+      # make file doesn't set rpath on mac yet
+      system "make", "PREFIX=#{libexec}", "EXTRA_LFLAGS=-Wl,-rpath,#{libexec}/lib"
+    else
+      # On Linux, also set RPATH in LIBFLAGS, so that the linker will also add the RPATH to .hdll files.
+      inreplace "Makefile", "LIBFLAGS =", "LIBFLAGS = -Wl,-rpath,${INSTALL_LIB_DIR}"
+      system "make", "PREFIX=#{libexec}"
+    end
```

Comment:
Can these rpath fixes be upstreamed?

Reply:
I'm not sure if the first (mac) fix would make sense to upstream. From what I can tell, it is needed because it seems that homebrew doesn't "fix up" the library paths for the HDLLs (hashlink-specific dynamic libraries) in `libexec/lib`, so there is [an error when trying to load them](https://github.com/Homebrew/homebrew-core/actions/runs/3281027525/jobs/5402611558#step:6:47). Adding the rpath seems to avoid that issue (possibly only because homebrew does some extra stuff afterwards?), but it seems more like a workaround than a proper solution, and it would not help on other mac systems which aren't using homebrew (as far as I can tell).

The second make fix on Linux is explained further here: https://github.com/Homebrew/homebrew-core/pull/99237#issuecomment-1121425545. In short, it is needed for homebrew because it has tests to check if all linked libraries are linked in a way that in all theoretical situations, they can be loaded successfully. Hashlink's .hdlls depend on libhl.dylib, however, the hdlls are only ever used by hashlink apps which have always already loaded libhl.dylib before loading .hdlls. This is why there has not been a reason for the hdlls to be linked "properly" to libhl.dylib so far, and so the makefile patch is only needed to avoid the homebrew test failure. However, I could try to make a PR to get it upstreamed.

Either way, the fixes would only be available in Hashlink 1.13 which probably won't be out for a while.